core/src/main/java/com/linecorp/armeria/common/HttpResponseBuilder.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    /**
+     * Sets the status for this response.
+     */
+    public HttpResponseBuilder status(HttpStatus status) {
+        requireNonNull(status, "status");
+        responseHeadersBuilder.status(status);
+        return this;
+    }
+
+    /**
+     * Sets 200 OK to the status of this response.
+     */
+    public HttpResponseBuilder ok() {
+        status(HttpStatus.OK);
+        return this;
+    }
+
+    /**
+     * Sets 201 Created to the status of this response.
+     */
+    public HttpResponseBuilder created() {
+        status(HttpStatus.CREATED);
+        return this;
+    }
+
+    /**
+     * Sets 301 Moved Permanently to the status of this response.
+     */
+    public HttpResponseBuilder movedPermanently() {
+        status(HttpStatus.MOVED_PERMANENTLY);
+        return this;
+    }
+
+    /**
+     * Sets 400 Bad Request to the status of this response.
+     */
+    public HttpResponseBuilder badRequest() {
+        status(HttpStatus.BAD_REQUEST);
+        return this;
+    }
+
+    /**
+     * Sets 401 Unauthorized to the status of this response.
+     */
+    public HttpResponseBuilder unauthorized() {
+        status(HttpStatus.UNAUTHORIZED);
+        return this;
+    }
+
+    /**
+     * Sets 403 Forbidden to the status of this response.
+     */
+    public HttpResponseBuilder forbidden() {
+        status(HttpStatus.FORBIDDEN);
+        return this;
+    }
+
+    /**
+     * Sets 404 Not Found to the status of this response.
+     */
+    public HttpResponseBuilder notFound() {
+        status(HttpStatus.NOT_FOUND);
+        return this;
+    }
+
+    /**
+     * Sets 500 Internal Server Error to the status of this response.
+     */
+    public HttpResponseBuilder internalServerError() {
+        status(HttpStatus.INTERNAL_SERVER_ERROR);
+        return this;
+    }
+
+    /**
+     * Sets 502 Bad Gateway to the status of this response.
+     */
+    public HttpResponseBuilder badGateway() {
+        status(HttpStatus.BAD_GATEWAY);
+        return this;
+    }
+
+    /**
+     * Sets the content as UTF_8 for this response.
+     */
+    public HttpResponseBuilder content(String content) {
+        requireNonNull(content, "content");
+        return content(MediaType.PLAIN_TEXT_UTF_8, content);
+    }
+
+    /**
+     * Sets the content for this response.
+     */
+    public HttpResponseBuilder content(MediaType contentType, CharSequence content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+        return content(contentType, HttpData.of(contentType.charset(StandardCharsets.UTF_8), content));
+    }
+
+    /**
+     * Sets the content for this response.
+     */
+    public HttpResponseBuilder content(MediaType contentType, String content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+        return content(contentType, HttpData.of(contentType.charset(StandardCharsets.UTF_8), content));
+    }
```

Comment:
We may extract the common part between `AbstractHttpRequestBuilder` and `HttpResponseBuilder` to `AbstractHttpMessageBuilder`.

Reply:
Would be useful if we add `contentJson(Object content)`?

Reply:
> We may extract the common part between AbstractHttpRequestBuilder and HttpResponseBuilder to AbstractHttpMessageBuilder.

Should I address it in this PR?

> Would be useful if we add contentJson(Object content)?

Reply:
How about also adding the `content` method that takes `Publisher` as an argument?
```
public HttpResponseBuilder content(Publisher<? extends HttpData> content) {...}
```

Reply:
> Should I address it in this PR?

I hope so. 😀 Because duplicate code is caused by this new feature.

Reply:
> How about also adding the content method that takes Publisher as an argument?

It seems to be nice.